### ISSUE TITLE
Replace custom fonts with system fonts

### ISSUE THREAD

**Issue by theskumar** [HyphaApp/hypha#3131]
**Is your feature request related to a problem? Please describe.**

Right now, public and apply sites use custom web fonts - 'Nato-sans' and 'montserrat' which add an extra 1.3M download size to the page when loaded initially. It adds to the overall server cost and requires extra bandwidth for the end user. 

Moreover,  it causes re-rendering of the page and adds a flickering effect to the site. 

**Describe the solution you'd like**

Replace custom fonts with the fonts available on the user's system. The wagtail site now also uses the `system-fonts`; this will allow for a consistent look across the site and also a familiar font experience. 

Popular websites use System Fonts, including:

- GitHub
- Ghost
- Bootstrap
- Booking.com

The new font stack will be, read more at https://flaviocopes.com/css-system-fonts/
```
-apple-system,BlinkMacSystemFont,Segoe UI,system-ui,Roboto,Helvetica
 Neue,Arial, sans-serif,Apple Color Emoji,Segoe UI Emoji,Segoe UI Symbol,Noto Color Emoji;
```
**Describe alternatives you've considered**

- Load the fonts over cdn, bandwidth reduction, but not much
- Use a JS-based font loader to avoid the flickering effect; display the content only after fonts are loaded.

**Additional context**

Screenshots from the mac-os rendering:

![Screenshot 2023-01-30 at 07 21 08@2x](https://user-images.githubusercontent.com/236356/215412579-26485d78-2efd-440e-b66e-00441ab0c2eb.jpg)
![Screenshot 2023-01-30 at 07 21 32@2x](https://user-images.githubusercontent.com/236356/215412639-ba628307-aded-4bee-aa78-885c5317adfa.jpg)

**Priority**

- Low priority (annoying, would be nice to see)

**Affected roles**

- Staff
- Applicants
- Reviewers
- Partners
- Developers
- General public

**Ideal deadline**

ASAP

**Comment by frjo** [comment]
I believe "system-ui" is safe to use nowadays, https://caniuse.com/font-family-system-ui.

**Comment by theskumar** [comment]
> I believe "system-ui" is safe to use nowadays; https://caniuse.com/font-family-system-ui.

Yes. `system-ui` is supported well. I went ahead with what wagtail has to be safer with people still having an older browser version. 

We can upgrade to use `system-ui` as the first font in the stack, while keeping `-apple-system, BlinkMacSystemFont` as a fallback.

or another option is to upgrade as the wagtail admin upgrades its font stack.

**Comment by theskumar** [comment]
As discussed over the call, we would keep the 'Nato-sans' and 'montserrat' fonts for the public site.
